### PR TITLE
Release 0.40.3

### PR DESCRIPTION
**Pull request by khoroshevskyi**
Small bug fixes
Changes:
- Fixed https://github.com/pepkit/peppy/issues/393
- Fixed https://github.com/pepkit/peppy/issues/480
- Fixed https://github.com/pepkit/peppy/issues/369
- Fixed https://github.com/pepkit/peppy/issues/399
- Fixed https://github.com/pepkit/peppy/issues/476
- Fixed https://github.com/pepkit/peppy/issues/471

### CODE REVIEW

File: peppy/sample.py [hunk 1]
```diff
@@ -134,31 +135,39 @@ def _obj2dict(obj, name=None):
...
-            _LOGGER.debug("Sample data written to: {}".format(path))
+        if path:
+            path = os.path.expandvars(path)
+            if os.path.exists(os.path.dirname(path)):
+                with open(path, "w") as outfile:
+                    try:
+                        yaml_data = yaml.safe_dump(serial, default_flow_style=False)
+                    except yaml.representer.RepresenterError:
+                        _LOGGER.error("Serialized sample data: {}".format(serial))
+                        raise
```
Comment (nleroy917) [inline peppy/sample.py:159]: What does this do?

File: peppy/simple_attr_map.py [hunk 2]
```diff
@@ -46,6 +46,9 @@ def __getattr__(self, item):
         except KeyError:
             raise AttributeError(f"Attribute not found: {item}")
 
+    def __eq__(self, other: "SimpleAttMap"):
+        return self._mapped_attr == other._mapped_attr
```
Comment (nleroy917) [inline peppy/simple_attr_map.py:50]: Tests for these? Is the `mapped_attr` just a dictionary?

**Review by nleroy917 — APPROVED** [review]
Looks great

File: peppy/simple_attr_map.py (hunk 2 above)
Comment (khoroshevskyi) [reply peppy/simple_attr_map.py:50]: Yeah, I added tests. it is just dict:)

File: peppy/sample.py (hunk 1 above)
Comment (khoroshevskyi) [reply peppy/sample.py:159]: saving config to yaml file

Comment (donaldcampbelljr) [reply peppy/sample.py:159]: It saves the Sample to yaml

**Review by donaldcampbelljr — APPROVED** [review]
Looks good!

**Comment by donaldcampbelljr** [comment]
I just realized that this needs to be release **v0.40.3**
We released **0.40.2** in May:
[v0.40.2](https://github.com/pepkit/peppy/releases/tag/v0.40.2)

**Pull request merged**